### PR TITLE
[WIP] Unit test sub-directories - Continued

### PR DESCRIPTION
**Pull request by ccdle12**
fixes #12583

Continuing on the work of @chrislennon.

I've applied his changes to the tests and also updated: 

`unit_test_directory(const std::string fileName)` 

* It now passes in the the current fileName to be used as part of the temp directory file path.

Any guidance on further improvements or refactored would be very much welcomed

### CODE REVIEW

File: src/test/test_bitcoin.cpp [hunk 1]
```diff
@@ -68,7 +68,7 @@ TestingSetup::TestingSetup(const std::string& chainName) : BasicTestingSetup(cha
 
         RegisterAllCoreRPCCommands(tableRPC);
         ClearDatadirCache();
-        pathTemp = fs::temp_directory_path() / strprintf("test_bitcoin_%lu_%i", (unsigned long)GetTime(), (int)(InsecureRandRange(100000)));
+        fs::path pathTemp = util_tests::unit_test_directory("test_bitcoin");
```
Comment (MarcoFalke) [inline src/test/test_bitcoin.cpp]: This shadows `pathTemp`, you might as well rename it `m_path_temp` according to the developer guidelines

File: src/test/util_tests.cpp [hunk 2]
```diff
@@ -21,6 +21,26 @@
...
 
+namespace util_tests {
+    // Create and return temporary subdirectory for test outputs
+    fs::path unit_test_directory(const std::string fileName){
+
+        std::string sub_dir = "bitcoin_unit_tests";
+        fs::path dir(fs::temp_directory_path() / sub_dir);
+        fs::path test_dir(dir / fileName / std::string(boost::unit_test::framework::current_test_case().p_name));
+
+        if(!fs::exists(dir))
```
Comment (MarcoFalke) [inline src/test/util_tests.cpp]: The dir will exist when the previous run was not shut down cleanly or when running two `test_bitcoin` at the same time. I don't think we should silently continue here.

**Comment by MarcoFalke** [comment]
Still needs rebase

**Comment by ccdle12** [comment]
Have really made a mess of this PR, appreciate help to remedy it but understand if it would be cleaner to scrap it all together

**Comment by MarcoFalke** [comment]
```bash
git checkout $your_branch
git merge 0bc980b1f65f3739bb9667d58abab1137c0f942a # merge all master changes
git reset --soft 0bc980b1f65f3739bb9667d58abab1137c0f942a # consider all non-master changes to be put in a commit
git commit -m 'commit message' # Put all non-master changes in this commit
git push origin $your_branch -f # Force push the rewritten branch

**Comment by ccdle12** [comment]
Currently stuck, would really appreciate any guidance on the following:

- Can compile bitcoind tests by running ```$ make -C src/test```
- Tests successfully complete by running the command - ```$ src/test/test_bitcoin``` and  a common temp folder is created - ```bitcoin_unit_tests```

Extending these tests to ```qt``` tests when running ```$ make``` at the root folder returns an error -
```
util_tests::unit_test_directory(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)", referenced from:
      _main in test_bitcoin_qt-test_main.o
      TestingSetup::TestingSetup(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in qt_test_test_bitcoin_qt-test_bitcoin.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

I've attempted to include ```test/utils.h``` in ```Makefile.qttest.include``` but that doesn't seem to be the correct solution

**Pull request closed without merging**